reverse_file.py

```diff
@@ -1 +1,53 @@
+import os
 
+
+def file_length(name: str) -> int:
+    length = os.stat(name).st_size
+    return length
+
+
+def read_file(name: str, offset: int, size: int, output_buf_offset: int) -> bytearray:
+    """
+    Reads `size` bytes from file `name` starting from `offset` index and
+    stores them into `output_buf` starting from `output_buf_offset`
+  """
+    with open(name, "rb") as f:
+        f.seek(offset)
+        chunk = f.read(size)
+        output_buf = bytearray(chunk[output_buf_offset:])
```

Comment:
The meaning of `output_buf_offset` is different (but it make sense only if `output_buf` is also provided as a parameter) -- it sets the position at which data is written into `output_but`, but the chunk is always written completely

Reply:
That's true, I haven't used this approach in python before but it makes sense